lib/http/2/connection.rb

```diff
@@ -226,6 +226,10 @@ def receive(data)
         else
           case frame[:type]
           when :headers
+            # When server receives even-numbered stream identifier,
+            # the endpoint MUST respond with a connection error of type PROTOCOL_ERROR.
+            connection_error if frame[:stream].even? && is_a?(Server)
```

Comment:
👍 

Minor nitpick (surprised rubocop didn't complain): https://github.com/georgeu2000/http-2/blob/f52efc656619f5f05b4f69e203f2ffd1ae4bd10e/lib/http/2/connection.rb#L638 - for consistency, can we self.is_a?